Sources/Honeycomb/HoneycombNavigationSpanInstrumentation.swift

```diff
@@ -6,84 +6,85 @@ private let navigationInstrumentationName = "@honeycombio/instrumentation-naviga
 private let navigationSpanName = "Navigation"
 private let unencodablePath = "<unencodable path>"
 
-private func activeBaggage() -> Baggage? {
-    return OpenTelemetry.instance.contextProvider.activeBaggage
-}
-
 func getTracer() -> Tracer {
     return OpenTelemetry.instance.tracerProvider.get(
         instrumentationName: navigationInstrumentationName,
         instrumentationVersion: honeycombLibraryVersion
     )
 }
 
-@available(iOS 16.0, macOS 12.0, *)
-func reportNavigation(path: NavigationPath) {
-    if let codablePath = path.codable {
-        reportNavigation(path: codablePath)
-    } else {
-        reportNavigation(path: unencodablePath)
-    }
+internal class HoneycombNavigationProcessor {
```

Comment:
I was expecting the `internal` keyword here to limit access to this class, but it looks like I could still call it from our smoke test app. 

ie. the following, in a view in the smoke test app, was **not** an error and compiled successfully:

```swift
HoneycombNavigationProcessor.singleton.reportNavigation(path: "asdf")
```

What am I missing in order to limit access to this singleton and force clients to go through the entrypoints in the main Honeycomb class?